pkg/github/search.go

```diff
@@ -86,7 +86,7 @@ func searchCode(client *github.Client, t translations.TranslationHelperFunc) (to
 				mcp.Description("Sort order ('asc' or 'desc')"),
 				mcp.Enum("asc", "desc"),
 			),
-			mcp.WithNumber("per_page",
+			mcp.WithNumber("perPage",
```

Comment:
I'm not totally opposed to this, but the reason some of these existed in snake case form was because that's [what the anthropic server exposed](https://github.com/modelcontextprotocol/servers/blob/d9f2cb0b1d32acec2d9f1d8d45cf8abaf24791b6/src/github/operations/search.ts#L8), and initially we just wanted to provide parity before moving forwards.

Obviously I introduced a few silly bugs in doing that last minute 😅

Reply:
Hm, they actually use a mix:
* `per_page`: https://github.com/modelcontextprotocol/servers/blob/d9f2cb0b1d32acec2d9f1d8d45cf8abaf24791b6/src/github/operations/search.ts#L8
* `perPage` https://github.com/modelcontextprotocol/servers/blob/d9f2cb0b1d32acec2d9f1d8d45cf8abaf24791b6/src/github/operations/commits.ts#L9

IMO, if that's not crucial it would be better to revert all to `per_page` because that's what GitHub API uses and to reduce confusion.

Reply:
> Hm, they actually use a mix

Right, so in #90 I changed the tool schemas to match the mix (but messed up the implementation).

I think for the moment let's keep this as `perPage`, and then we can talk about broader design principles at another time, looking at the entire API surface area. There's definitely a bit of tension between consistency within github-mcp-server, with the JSON-RPC envelope (all camel cased) with the GitHub API (mostly/all snake cased?).

I talked to the team and we're happy to get this in, I'm just going to take a closer look and maybe add some tests to avoid this in future. Thanks!